src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -400,46 +432,50 @@ def _check_value(self, action, value):  # pylint: disable=too-many-statements, t
                                 import subprocess
                                 import platform
                                 exit_code = subprocess.call(cmd_list, shell=platform.system() == 'Windows')
-                                telemetry.set_user_fault("Extension {} dynamically installed and commands will be "
-                                                         "rerun automatically.".format(ext_name))
                                 self.exit(exit_code)
                             else:
-                                error_msg = 'Extension {} installed. Please rerun your command.'.format(ext_name)
+                                with CommandLoggerContext(logger):
+                                    reminder = 'Extension {} installed. Please rerun your command.'.format(ext_name)
+                                    logger.error(reminder)
+                                self.exit(2)
```

Comment:
Tested the code. The result of telemetry is None when the extension is dynamically installed, whether run_after_extension_installed is true or false.

Reply:
That's the same problem with the above one.
Will set userfault for both cases.

Reply:
Tested.